modules/core/src/smithy4s/http/HttpUnaryClientCodecs.scala

```diff
@@ -119,19 +119,10 @@ object HttpUnaryClientCodecs {
 
       val mediaTypeWriters = new CachedSchemaCompiler.Uncached[HttpRequest.Writer[Blob, *]] {
         def fromSchema[A](schema: Schema[A]): HttpRequest.Writer[Blob, A] = {
-          val maybeRawMediaType = HttpMediaType.fromSchema(schema).map(_.value)
-          maybeRawMediaType match {
-            case Some(mt) =>
-              new HttpRequest.Writer[Blob, A] {
-                def write(request: HttpRequest[Blob], value: A): HttpRequest[Blob] =
-                  request.withContentType(mt)
-              }
-            case None =>
-              new HttpRequest.Writer[Blob, A] {
-                def write(request: HttpRequest[Blob], value: A): HttpRequest[Blob] =
-                  if (request.body.isEmpty) request
-                  else request.withContentType(requestMediaType)
-              }
+          new HttpRequest.Writer[Blob, A] {
+            def write(request: HttpRequest[Blob], value: A): HttpRequest[Blob] =
+              if (request.body.isEmpty) request
+              else request.withContentType(requestMediaType)
```

Comment:
**thought:** I am not convinced about this change. 
I added following test in alloy:
```smithy
apply HttpPayloadWithDefault @httpRequestTests([{
    id: "SimpleRestJsonSomeHttpPayloadWithDefault"
    documentation: "Pass simple string value as is if payload provided"
    protocol: simpleRestJson
    method: "PUT"
    uri: "/httpPayloadWithDefault"
    body: "\"custom value\""
    headers: {
        "Content-Type": "application/json"
    }
    requireHeaders: ["Content-Length"]
    params: {
        nested: "custom value"
    }
    bodyMediaType: "application/json"
}])
```
and without this change the test fails in one case:
```
[error] [http4s] - alloy.test#HttpPayloadWithDefault(client|Request): SimpleRestJsonSomeHttpPayloadWithDefault 3ms
[error] [http4s]   Header Content-Type has value `text/plain` but expected `application/json` (modules/tests/src/smithy4s/tests/ProtocolComplianceSuite.scala:183)
[error] [http4s]
[error] [http4s]     [INFO] 15:03:03 [ProtocolComplianceSuite.scala:173] Pass simple string value as is if payload provided
[error] [http4s]
```